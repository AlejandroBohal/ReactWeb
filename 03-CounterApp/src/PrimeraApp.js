import React from 'react';
import PropTypes from 'prop-types';
// Functional components
const PrimeraApp = ( {saludo,subtitulo} ) => {
    
    return( 
        <>
            {/*<pre>{ JSON.stringify(saludo,null,3) }</pre>*/}
            <h2>{saludo}</h2>
            <p>{subtitulo}</p>
            
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo',
}
export default PrimeraApp;