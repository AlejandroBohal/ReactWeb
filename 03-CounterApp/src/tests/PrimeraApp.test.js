import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
import React from 'react';


describe('PrimeraApp component test', () => {
    //test('should show "Adios mundo" message', () => {
    //    
    //    const saludo = "Adios mundo";
    //    const {getByText} = render(<PrimeraApp saludo = {saludo} />);
    //    expect(getByText(saludo)).toBeInTheDocument();
    //})
    test('should show <PrimeraApp /> correctly', () => {
        const saludo = 'Hola, soy Goku'
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should show subtitulo sent by props', () => {
        const saludo = 'Hola, soy Goku'
        const subtitulo = 'Soy una prueba'
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo ={subtitulo} />);
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);

    });
    
    
    
})
