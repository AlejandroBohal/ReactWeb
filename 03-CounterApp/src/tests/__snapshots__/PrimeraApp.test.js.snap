// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp component test should show <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`PrimeraApp component test should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h2>
    Hola, soy Goku
  </h2>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
